### DISCUSSION TITLE
Using Red/Black Tree

### DISCUSSION THREAD

**Discussion by saleh-unikie** [zephyrproject-rtos/zephyr#39310] [Q&A]
Hi

I am going to use red/black tree provided as a part of the Zephyr project:
https://docs.zephyrproject.org/latest/reference/data_structures/rbtree.html

But I don't have any idea about how to use it. Can anyone give a simple example? For example if we have a set of integers and we want to store them in a red/black tree, how it can be done?

Thanks in advance!

**Comment by yashi** [comment]
How about a sample code in Zephyr tree?
https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/unit/rbtree/main.c
